From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 25 Apr 2024 14:48:23 -0500
Subject: [PATCH] Fix test_empty_NO_COLOR_and_FORCE_COLOR_ignored when
 TERM=dumb

Dumb terminals are incapable of handling color, so this test would fail
---
 testing/io/test_terminalwriter.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/testing/io/test_terminalwriter.py b/testing/io/test_terminalwriter.py
index afa8d5cae..d4b4786dd 100644
--- a/testing/io/test_terminalwriter.py
+++ b/testing/io/test_terminalwriter.py
@@ -221,7 +221,10 @@ def test_NO_COLOR_and_FORCE_COLOR(
     monkeypatch.setitem(os.environ, "FORCE_COLOR", FORCE_COLOR)
     assert_color(expected)
 
-
+@pytest.mark.skipif(
+    os.environ["TERM"] == "dumb" or "NO_COLOR" in os.environ,
+    reason="dumb terminals can't handle color"
+    )
 def test_empty_NO_COLOR_and_FORCE_COLOR_ignored(monkeypatch: MonkeyPatch) -> None:
     monkeypatch.setitem(os.environ, "NO_COLOR", "")
     monkeypatch.setitem(os.environ, "FORCE_COLOR", "")
